### DISCUSSION TITLE
Vertical Bar content orientation

### DISCUSSION THREAD

**Discussion by hastiin-manuelito** [qtile/qtile#3506] [General]
After a brief exploration of the Qtile website, and a quick search of "Bar" in this discussion forum I did not find anything about widgets within the vertical bar orientations that print text horizontally (no args or kwargs, for example).

Is it possible, within the context of the libqtile libraries, to make it so that the text in the vertical orientations output horizontally? I run a two screen setup, and I would like the secondary screen (my smaller laptop screen) to display state information on the side. Sometimes I just like my active screen to be completely blank, no clutter whatsoever, especially when reading. Thank you in advance for any responses and/or suggestions!

A screenshot of a rough idea of what I am trying to achieve and what is.

![2022-04-14-131505_1918x1080_scrot](https://user-images.githubusercontent.com/78440845/163453038-1ef6be46-ab61-4aa6-b623-44bfdcf42e46.png)

**Comment by elParaguayo** [comment]
In short, no. There's no current way to set the text direction. This is done automatically based on the bar's orientation.

**Comment by hastiin-manuelito** [comment]
@elParaguayo Thanks for the prompt response!

If you may indulge me for a bit...

Were I to attempt to code such a feature in my build, generally speaking, what possible ways might I achieve this?

Two scenarios that come to mind:
 1. configuring the "right" bar (or "left") as a series of horizontally stacked "Bar" classes each with one widget of its own.
 2. "Rotating" the individual widgets within the Bar instance (or Gap), and having them stacked above one another.

Perhaps you can comment on how much effort and code either of these (or alternatives) might take to implement in my build, and perhaps make available to the community, if it is worth incorporating.

Thanks for your time.

**Reply by elParaguayo** [reply]
Ok. So, it's absolutely possible to do this but would involve a fair bit of rewriting.

Bar's work by positioning widgets based on their lengths. For vertical bars, the length of a widget is its size from top to bottom. In addition, widgets have specific appearance/layout logic depending on whether the bar is vertical or horizontal.

So, to do what you want to do:
* You an option to override widgets' logic so that they are still displayed horizontally in a vertical bar (you may need to fix the widget's width to the width of the bar)
* You need to bar to calculate the offset for each widget using length (if normal) or the height if you're trying to show widget's horizontally in a vertical bar.

It's probably a fair bit of effort, especially if you're not already familiar with how the bar and widgets work. However, it's definitely doable.

I will add that I am hoping to update my [popup toolkit](https://qtile-extras.readthedocs.io/en/latest/manual/how_to/popup.html) to be able to use widgets. If I ever get round to doing that then you could create a window like you want and place it at the side of your screen (using a `Gap` to reserve some screen scpace).
